src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -7305,10 +7305,24 @@ export class StudyViewPageStore {
                 }
 
                 if (!_.isEmpty(this.sampleTreatments.result)) {
-                    ret['SAMPLE_TREATMENTS'] = 1;
+                    ret['SAMPLE_TREATMENTS'] = this.sampleTreatments.result
+                        ?.map(row => row.samples)
+                        .reduce((allSamples, samples) => {
+                            samples.forEach(s =>
+                                allSamples.add(s.sampleId + '::' + s.studyId)
+                            );
+                            return allSamples;
+                        }, new Set<String>()).size as any;
                 }
                 if (!_.isEmpty(this.patientTreatments.result)) {
-                    ret['PATIENT_TREATMENTS'] = 1;
+                    ret['PATIENT_TREATMENTS'] = this.patientTreatments.result
+                        ?.map(row => row.samples)
+                        .reduce((allSamples, samples) => {
+                            samples.forEach(s =>
+                                allSamples.add(s.sampleId + '::' + s.studyId)
+                            );
+                            return allSamples;
+                        }, new Set<String>()).size as any;
```

Comment:
as discussed maybe good to make this a function just to make it clear they are the same apart from using `this.sampleTreatments.result` vs `this.patientTreatments.result`